homeassistant/components/klyqa/datacoordinator.py

```diff
@@ -0,0 +1,224 @@
+"""Klyqa datacoordinator."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+import socket
+from typing import Any
+
+from klyqa_ctl import klyqa_ctl as api
+
+from homeassistant.const import (
+    CONF_HOST,
+    CONF_PASSWORD,
+    CONF_SCAN_INTERVAL,
+    CONF_USERNAME,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_component import (
+    DEFAULT_SCAN_INTERVAL,
+    EntityComponent,
+)
+
+from .const import (
+    CONF_POLLING,
+    CONF_SYNC_ROOMS,
+    EVENT_KLYQA_NEW_LIGHT,
+    EVENT_KLYQA_NEW_LIGHT_GROUP,
+    LOGGER,
+)
+
+
+class HAKlyqaAccount(api.Klyqa_account):  # type: ignore[misc]
+    """HAKlyqaAccount."""
+
+    hass: HomeAssistant | None
+
+    udp: socket.socket
+    tcp: socket.socket
+    polling: bool
+    sync_rooms: bool
+    scan_interval_conf: float
+
+    def __init__(
+        self,
+        udp: Any,
+        tcp: Any,
+        username: str = "",
+        password: str = "",
+        host: str = "",
+        hass: HomeAssistant | None = None,
+        polling: bool = True,
+        sync_rooms: bool = True,
+        scan_interval: float = -1.0,
+    ) -> None:
+        """HAKlyqaAccount."""
+        super().__init__(username, password, host)
+        self.hass = hass
+        self.udp = udp
+        self.tcp = tcp
+        self.polling = polling
+        self.sync_rooms = sync_rooms
+        self.scan_interval_conf = scan_interval
+
+    async def send_to_bulbs(
+        self,
+        args_parsed: Any,
+        args_in: Any,
+        timeout_ms: Any = 5000,
+        async_answer_callback: Any = None,
+    ) -> Any:
+        """Send_to_bulbs."""
+        ret = await super()._send_to_bulbs(
+            args_parsed,
+            args_in,
+            self.udp,
+            self.tcp,
+            timeout_ms=timeout_ms,
+            async_answer_callback=async_answer_callback,
+        )
+        return ret
+
+    # pylint: disable=arguments-differ
+    async def login(self, **kwargs: Any) -> bool:
+        """Login."""
+        ret: bool = await super().login(**kwargs)
+        if ret:
+            await api.async_json_cache(
```

Comment:
I thought I would offer a user data cache for when the user wants to make a change in his configuration and therefore deletes its previous account and when creating then a new one he gets his previous data predefined so he can choose if he wants to keep this data just change a slight bit of it and then recreates his integration login. But as you want to forbid a user defined room sync or update interval, this can be remove as well I guess.
And of course therefore is the option flow code, I know that, but I haven't coded it finally yet.